### PR TITLE
Upgrade to Python 3.10

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/959
Upgrade Python from 3.9 to 3.10

Notes:
- need to switch from the library orderedset to ordered_set since the former does not support 3.10 and is possibly abandonware.

# Test instructions | Instructions pour tester la modification

See if this utils works with upgraded apps:
admin: https://github.com/cds-snc/notification-admin/pull/1400
api:
document-download-api:

### CODE REVIEW

**Review by whabanks — COMMENTED** [review]
Looks good, just one question: Should we update `pyproject.toml` as well for the black formatter?
```
[tool.black]
line-length = 130
target-version = ['py37', 'py38', 'py39'] # Add 'py310'
```

File: .github/actions/waffles/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 docopt==0.6.2
 Flask==2.0.3
 markupsafe==2.0.1
-git+https://github.com/cds-snc/notifier-utils.git@48.3.0#egg=notifications-utils
+# git+https://github.com/cds-snc/notifier-utils.git@48.3.0#egg=notifications-utils
+git+https://github.com/cds-snc/notifier-utils.git@python-3.10#egg=notifications-utils
```
Comment (smcmurtry) [inline .github/actions/waffles/requirements.txt:5]: I guess we will need a follow up PR to point this to a tagged version of utils? Also - this won't work if the `python-3.10` branch is auto deleted when this PR is merged. So maybe we should temporarily turn the "auto delete branches" feature off?

**Review by smcmurtry — APPROVED** [review]
I tested this with your [admin PR](https://github.com/cds-snc/notification-admin/pull/1400) doing the following:
- creating SMS and email templates
- sending one off SMS and emails
- csv sending emails

And I didn't find any bugs. The tests are all passing in the admin PR so that also tells me the utils changes are working. 

I also wanted to check that the new `ordered_set` library was working, so I found a place in admin that uses that code from the `Field` class - `app.main.validators.NoCommasInPlaceHolders`. The validation error from that code is working as expected:

Screen Shot 2022-11-28 at 5 04 49 PM

File: .github/actions/waffles/requirements.txt (hunk 1 above)
Comment (sastels) [reply .github/actions/waffles/requirements.txt:5]: good point. I'll do it all in this PR I think, ie change waffles to point to 50.0.0 and then tag quickly after the merge. Should work!

**Comment by sastels** [comment]
> Looks good, just one question: Should we update the black formatter config `pyproject.toml` as well?
> 
> ```
> [tool.black]
> line-length = 130
> target-version = ['py37', 'py38', 'py39'] # Add 'py310'
> ```

Oh, good catch! I'll see about updating this before merging.

**Pull request merged**